crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -121,15 +123,47 @@ impl Plugin for VisibilityPlugin {
 pub fn calculate_bounds(
     mut commands: Commands,
     meshes: Res<Assets<Mesh>>,
-    without_aabb: Query<(Entity, &Handle<Mesh>), (Without<Aabb>, Without<NoFrustumCulling>)>,
+    without_aabb_or_with_changed_mesh: Query<
+        (Entity, &Handle<Mesh>),
+        (
+            Or<(Without<Aabb>, Changed<Handle<Mesh>>)>,
+            Without<NoFrustumCulling>,
+        ),
+    >,
+    mut entity_mesh_map: Local<HashMap<Handle<Mesh>, Vec<Entity>>>,
+    mut mesh_events: EventReader<AssetEvent<Mesh>>,
 ) {
-    for (entity, mesh_handle) in without_aabb.iter() {
+    for (entity, mesh_handle) in without_aabb_or_with_changed_mesh.iter() {
+        // Record entities that have mesh handles.
+        entity_mesh_map
+            .entry(mesh_handle.clone())
```

Comment:
How cheap is cloning the `mesh_handle`? If cloning is more expensive than hashing we could presumably do a `contains`+`insert` check instead of `entry` :balance_scale:

Reply:
Extremely cheap: they're basically just a `Rc` smart pointer.

Reply:
yes!
`clone` is cheap but is reference counted, `clone_weak` isn't so it's quite cheaper

Reply:
Updated to `clone_weak` in 92e78bc2d90c09db4fc4502efa2d8ae22c311753.